synapse/storage/databases/main/purge_events.py

```diff
@@ -390,7 +390,6 @@ def _purge_room_txn(self, txn, room_id: str) -> List[int]:
             "event_search",
             "events",
             "group_rooms",
-            "public_room_list_stream",
```

Comment:
the fact this was here means that if you:

 * roll forward to synapse 1.52 (?)
 * then roll back again
 * then do a "delete room" via the admin API

... then it won't work, and you'll get a 500 error. I feel like that's a sufficiently unlikely combo that it's not worth worrying about, but if it is of concern then we could do another round of removing references (together with a `TRUNCATE TABLE`) and do the actual deletion in the future.